File: dcp-solutions/2024-11-19/solution.php
```php
<?php

class Solution {
    public function nthPerfectNumber($n) {
        $i = 1;
        $cnt = 0;
        while ($cnt < $n) {
            $num = (int)sqrt(10 * $i - 9);
            if ((int)sqrt((10 * $i - 9)) == (10 * $i - 9)) {
                $sum = array_sum(str_split($i));
                if ($sum == 10) {
                    $cnt++;
                    $result[] = $i;
                }
            } else {
                $i++;
            }
        }
        return end($result);
    }
}

?>
```

File: dcp-solutions/2024-11-19/solution.c
```c
#include <stdio.h>
#include <math.h>

long long nth_perfect_number(int n) {
    int i = 1;
    while (n > 0) {
        long long num = (long long)i * (i + 1);
        if ((num / 2 - i) == 10) {
            n -= 1;
        }
        i += 1;
    }
    return num;
}
```